### PR TITLE
Fix few --use-current-year bugs

### PR DESCRIPTION
**Pull request by aostrowski**
Parse end years correctly if extra spacing is used. 
Stop searching for the year when the whole license header was searched. 
Stop changing four digit numbers that come after the header.
Fail the check if the year update failed.
Avoid creating a range with the current year used as both start and end.
Add tests for cases when the header already contains the current year.

### CODE REVIEW

File: pre_commit_hooks/insert_license.py [hunk 1]
```diff
@@ -262,6 +267,7 @@ def license_not_found(  # pylint: disable=too-many-arguments
 def try_update_year_range(
     src_file_content: list[str],
     license_header_index: int,
+    license_lines: int
```
Comment (Lucas-C) [inline pre_commit_hooks/insert_license.py:270]:
```suggestion
    license_length: int
```

File: tests/resources/module_with_spaced_year_range_in_license.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Copyright (C) 2015 - 2017 Teela O'Malley
```
Comment (Lucas-C) [inline tests/resources/module_with_spaced_year_range_in_license.py:1]: Could you also please add a unit test for this fix:

> Stop changing four digit numbers that come after the header.

File: pre_commit_hooks/insert_license.py [hunk 3]
```diff
@@ -274,22 +280,22 @@ def try_update_year_range(
...
+            end_year = match[5:].lstrip(" -,")
+            if not end_year and start_year < current_year or end_year and int(end_year) < current_year:
                 updated = line.replace(match,
-                                    str(start_year) + '-' + str(current_year))
+                                       str(start_year) + '-' + str(current_year))
                 # verify the current list of years ends in the current one
                 if _YEARS_PATTERN.findall(updated)[-1][-4:] != str(current_year):
-                    print(f"Unable to update year range in line: {line.rstrip()}. Got: {updated.rstrip()}")
-                    break
+                    raise RuntimeError(f"Unable to update year range in line: {line.rstrip()}. Got: {updated.rstrip()}")
```
Comment (Lucas-C) [inline pre_commit_hooks/insert_license.py:295]:
```suggestion
                    raise RuntimeError(f"Year range detected in license header, but we were unable to update it on line: {line.rstrip()}. Got: {updated.rstrip()}")
```
Also, have you tested if the name of the file is mentioned in the final error mesage reported by `pre-commit`?
That would be very useful :)

File: pre_commit_hooks/insert_license.py [hunk 4]
```diff
@@ -168,14 +169,18 @@ def process_files(args, changed_files, todo_files, license_info: LicenseInfo):
...
+                if license_found(remove_header=args.remove_header,
+                                 update_year_range=args.use_current_year,
+                                 license_header_index=license_header_index,
+                                 license_info=license_info,
+                                 src_file_content=src_file_content,
+                                 src_filepath=src_filepath,
+                                 encoding=encoding):
+                    changed_files.append(src_filepath)
+            except RuntimeError as error:
+                print(error)
```
Comment (Lucas-C) [inline pre_commit_hooks/insert_license.py:182]: I'm not fan of this error-catching strategy:
you have zero ensurance that the error caught will be related to the year update operation!
Please either create a custom exception class or move this try-catch around the call to `try_update_year_range()` and add a boolean return value to `license_found()`

**Review by Lucas-C — CHANGES_REQUESTED** [review]
Thank you for coming back with some fixes!

File: tests/resources/module_with_spaced_year_range_in_license.py (hunk 2 above)
Comment (aostrowski) [reply tests/resources/module_with_spaced_year_range_in_license.py:1]: This is handled in the module_with_license_and_numbers.py file.

Comment (Lucas-C) [reply tests/resources/module_with_spaced_year_range_in_license.py:1]: Oh, alright. I was expecting a similar `"2015 - 2017"` string to appear in the code

File: pre_commit_hooks/insert_license.py (hunk 1 above)
Comment (aostrowski) [reply pre_commit_hooks/insert_license.py]: Done

File: pre_commit_hooks/insert_license.py (hunk 3 above)
Comment (aostrowski) [reply pre_commit_hooks/insert_license.py]: Good idea. I added a mechanism to capture and inspect stdout.

File: pre_commit_hooks/insert_license.py (hunk 4 above)
Comment (aostrowski) [reply pre_commit_hooks/insert_license.py:188]: I went with a custom exception class. I'm not a fan of returning just two booleans from a single function.

**Comment by aostrowski** [comment]
Thanks for the excellent suggestions :) I think I addressed all of them.

**Comment by Lucas-C** [comment]
Merged!
Good job @aostrowski, and thank you 😊

**Comment by Lucas-C** [comment]
Released: https://github.com/Lucas-C/pre-commit-hooks/releases/tag/v1.4.2

**Pull request merged**